src/Avalonia.Controls/Grid.cs

```diff
@@ -23,18 +23,54 @@ public class Grid : Panel
 
-            IsSharedSizeScopeProperty.Changed.AddClassHandler<Control>(DefinitionBase.OnIsSharedSizeScopePropertyChanged);
-            ColumnProperty.Changed.AddClassHandler<Control>(OnCellAttachedPropertyChanged);
-            ColumnSpanProperty.Changed.AddClassHandler<Control>(OnCellAttachedPropertyChanged);
-            RowProperty.Changed.AddClassHandler<Control>(OnCellAttachedPropertyChanged);
-            RowSpanProperty.Changed.AddClassHandler<Control>(OnCellAttachedPropertyChanged);
+            IsSharedSizeScopeProperty.Changed.AddClassHandler((Action<Control, AvaloniaPropertyChangedEventArgs>)DefinitionBase.OnIsSharedSizeScopePropertyChanged);
+            ColumnProperty.Changed.AddClassHandler((Action<Control, AvaloniaPropertyChangedEventArgs>)OnCellAttachedPropertyChanged);
+            ColumnSpanProperty.Changed.AddClassHandler((Action<Control, AvaloniaPropertyChangedEventArgs>)OnCellAttachedPropertyChanged);
+            RowProperty.Changed.AddClassHandler((Action<Control, AvaloniaPropertyChangedEventArgs>)OnCellAttachedPropertyChanged);
+            RowSpanProperty.Changed.AddClassHandler((Action<Control, AvaloniaPropertyChangedEventArgs>)OnCellAttachedPropertyChanged);
 
-            AffectsParentMeasure<Grid>(ColumnProperty, ColumnSpanProperty, RowProperty, RowSpanProperty);
+            AffectsParentMeasure<Grid>(ColumnProperty, ColumnSpanProperty, RowProperty, RowSpanProperty, ColumnDefinitionsProperty, RowDefinitionsProperty);
+            AffectsMeasure<Grid>(ColumnDefinitionsProperty, RowDefinitionsProperty);
 
+            RowDefinitionsProperty.Changed.AddClassHandler<Grid>(RowDefinitionsPropertyChanged);
+            ColumnDefinitionsProperty.Changed.AddClassHandler<Grid>(ColumnDefinitionsPropertyChanged);
         }
 
+        private static void ColumnDefinitionsPropertyChanged(Grid arg1, AvaloniaPropertyChangedEventArgs arg2)
+        {
+            var val = arg2.NewValue as ColumnDefinitions;
+            ColumnDefinitionsPropertyChangedCore(arg1, val);
+        }
+
+        private static void ColumnDefinitionsPropertyChangedCore(Grid arg1, ColumnDefinitions arg2)
+        {
+            if (arg1._data is null)
+                arg1._data = new ExtendedData();
+
+            arg1._data!.ColumnDefinitions = arg2;
+            arg1._data!.ColumnDefinitions.IsDirty = true;
+            arg1._data!.ColumnDefinitions.Parent = arg1;
+        }
+
+        private static void RowDefinitionsPropertyChanged(Grid arg1, AvaloniaPropertyChangedEventArgs arg2)
+        {
+            var val = arg2.NewValue as RowDefinitions;
+            RowDefinitionsPropertyChangedCore(arg1, val);
+        }
+
+        private static void RowDefinitionsPropertyChangedCore(Grid arg1, RowDefinitions arg2)
+        {
+            if (arg1._data is null)
+                arg1._data = new ExtendedData();
+
+            arg1._data!.RowDefinitions = arg2;
```

Comment:
I believe "!" is not required here. At least with latest VS Preview that I have installed.

Reply:
It complains here in Omnisharp though